builder/js/ncf.js

```diff
@@ -639,12 +657,21 @@ app.controller('ncf-builder', function ($scope, $modal, $http, $log, $location,
     var technique = angular.copy($scope.selectedTechnique);
     var origin_technique = angular.copy($scope.originalTechnique);
 
-    var data = { "path" :  $scope.path, "technique" : $scope.toTechNcf(technique) }
-
+    // transform technique so it is valid to send to API:
+    technique = toTechNcf(technique);
+    var data = { "path" :  $scope.path, "technique" : technique }
     // Function to use after save is done
     // Update selected technique if it's still the same technique
     // update technique from the tree
     var saveSuccess = function(data, status, headers, config) {
+      // Transform back ncfTechnique to UITechnique, that will make it ok
+      technique = toTechUI(technique);
```

Comment:
could you use another name ? otherwise we won't know when we are handling a techNCF or techUI